src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -69,10 +69,20 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
-        private void OnPackagesChanged(object sender, EventArgs e) {
+        private void OnBeforePackagesInstalled(object sender, EventArgs e) {
+            try {
+                _host.StopSessionAsync().Wait(3000);
```

Comment:
It is interesting to check what will happen if `StopSessionAsync` throw exception